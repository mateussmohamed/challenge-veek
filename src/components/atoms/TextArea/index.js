import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledTextArea = styled.textarea`
  width: 100%;
  height: 120px;
  padding: 10px;
  border-radius: 8px;
  border: 1px solid #d5d5d5;
  color: #000;
  font-size: 18px;
  font-weight: bold;
  resize: none;

  :active,
  :focus {
    box-shadow: 1px 1px 10px 0px rgba(235, 91, 73, 0.5);
    outline: 0;
  }
`

const TextArea = props => <StyledTextArea {...props} />

TextArea.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
}

export default TextArea
