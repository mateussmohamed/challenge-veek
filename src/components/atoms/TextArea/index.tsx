import React from 'react'
import styled from 'styled-components'

const StyledTextArea = styled.textarea<HTMLTextAreaElement>`
  width: 100%;
  height: 120px;
  padding: 10px;
  border-radius: 8px;
  border: 1px solid #d5d5d5;
  color: #000;
  font-size: 18px;
  font-weight: bold;
  resize: none;

  :active,
  :focus {
    box-shadow: 1px 1px 10px 0px rgba(235, 91, 73, 0.5);
    outline: 0;
  }
`

export const TextArea = props => <StyledTextArea {...props} />
