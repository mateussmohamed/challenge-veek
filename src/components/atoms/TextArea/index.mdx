---
name: TextArea
menu: Atoms
---

import { Playground, PropsTable } from 'docz'
import TextArea from './index'

# TextArea
<PropsTable of={TextArea} />

<Playground>
  <TextArea name="comments" value="bla bla bla bla" />
</Playground>
