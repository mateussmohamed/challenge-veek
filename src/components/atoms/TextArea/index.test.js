import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import TextArea from './index'

describe('[<TextArea />]', () => {
  const wrap = (props = {}) => shallow(<TextArea {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({ name: 'comments', value: 'bla bla bla' })

    expect(wrapper).toMatchSnapshot()
  })
})
