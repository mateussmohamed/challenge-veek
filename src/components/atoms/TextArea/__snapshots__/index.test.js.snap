// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<TextArea />] matches to snapshot 1`] = `
<TextArea__StyledTextArea
  name="comments"
  value="bla bla bla"
/>
`;
